### PR TITLE
Feat/event category pages

### LINKED ISSUE #165: Feed: Event Category Pages
When we click into once of these sections we should open a page of events that fall into the specified category

Image

Image

### PR DESCRIPTION
**Pull request by AlenGanopolsky**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/165)
Added the event-category page for each category

# How Has This Been Tested?
Manual testing - screen recordings shown below

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

https://github.com/user-attachments/assets/6390167d-035d-4724-9993-58ed9cd1054b

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: .claude/settings.local.json
Comment (adescoteaux1) [inline .claude/settings.local.json:1]: remove

File: frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        <Text
+          style={{
+            fontFamily: FontFamilies.bold,
+            fontSize: FontSizes.lg,
+            color: AppColors.primaryText,
+            paddingHorizontal: 20,
+            marginTop: 20,
+          }}
+        >
+          No events found for this category.
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx]: translate this

File: frontend/apps/mobile/components/EventCategoriesListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+export function EventCategoriesListItem({
+  eventOccurrence,
+  onPress,
+}: EventCategoriesListItemProps) {
+  const { location, event } = eventOccurrence;
+  const organization = useGetOrganization(event.organization_id);
+  const orgName = organization.data?.status === 200 ? organization.data.data.name : "";
+  const address = `${location.address_line1}, ${location.district} ${location.postal_code}`;
+  const ageRange = `Ages ${event.age_range_min} - ${event.age_range_max}`;
```
Comment (adescoteaux1) [inline frontend/apps/mobile/components/EventCategoriesListItem.tsx]: make sure "Ages" gets translated

File: frontend/apps/mobile/components/EventCategoriesListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <TouchableOpacity
+      onPress={onPress}
+      activeOpacity={0.7}
+      style={{
+        flexDirection: "row",
```
Comment (adescoteaux1) [inline frontend/apps/mobile/components/EventCategoriesListItem.tsx]: styling to nativewind classname prop

File: frontend/apps/mobile/contexts/auth-context.tsx [hunk 4]
```diff
@@ -149,12 +148,12 @@ export function AuthProvider({ children }: { children: ReactNode }) {
       },
       {
         onSuccess: async (resp: signupGuardianResponse) => {
+          console.log('success!');
```
Comment (adescoteaux1) [inline frontend/apps/mobile/contexts/auth-context.tsx]: remove

File: frontend/apps/web/Dockerfile [hunk 5]
```diff
@@ -12,8 +12,7 @@ COPY scripts ./scripts
 RUN bun install
 # Build the web app
 WORKDIR /app/apps/web
-
-RUN bun run build
+RUN bun x tsc -b && bun x vite build
```
Comment (adescoteaux1) [inline frontend/apps/web/Dockerfile]: undo change

**Review by adescoteaux1 — COMMENTED** [review]
only thing - can we update the images on the base feed to be set as these images instead of pulling from events

Image

**Comment by AlenGanopolsky** [comment]
> only thing - can we update the images on the base feed to be set as these images instead of pulling from events
> 
> Image



> only thing - can we update the images on the base feed to be set as these images instead of pulling from events
> 
> Image

Here is the updated version: 
image

File: frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  const theme = Colors.light;
+
+  const { t: translate } = useTranslation();
+  const category = params.category as string;
+  const { lat: geoLocationLat, lng: geoLocationLong } = useGeolocation();
+
+  const { data: localizedOccurrencesResp, isLoading } = useGetAllEventOccurrences({
+    category,
+    
+    lat: "13.7563",
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx]: Replace hard coded lat and long

File: frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  const displayCategory = category.charAt(0).toUpperCase() + category.slice(1);
+
+  return (
+    <ThemedView className="flex-1" style={{ paddingTop: insets.top }}>
+      <Stack.Screen options={{ headerShown: false }} />
+
+      {/* Header */}
+      <View style={{ paddingHorizontal: 20, paddingTop: 10, paddingBottom: 16, overflow: "hidden" }}>
+        {/* Background logos */}
+        <Image
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx]: Remove this, not implemented correctly and it will be a different ticket

File: frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+            left: -30,
+            top: 20,
+            width: 160,
+            height: 160,
+            opacity: 0.06,
+          }}
+          contentFit="contain"
+        />
+        <TouchableOpacity
+          onPress={() => router.canGoBack() ? router.back() : router.navigate("/(app)/(tabs)")}
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/event-categories/index.tsx]: When would you ever not be able to go back? Should be an impossible state for a "normal" user no?

File: frontend/apps/mobile/app/(app)/(tabs)/_layout.tsx [hunk 9]
```diff
@@ -25,7 +25,6 @@ export default function TabLayout() {
         name="profile"
         options={{ title: translate("nav.profile") }}
       />
-      <Tabs.Screen name="org" options={{ href: null }} />
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/_layout.tsx:28]: Why was this removed?

File: frontend/apps/mobile/app/(app)/event/[id].tsx [hunk 10]
```diff
@@ -26,11 +27,14 @@ import { formatLocation } from "@/utils/format";
...
 function EventOccurrenceDetail({
   occurrence,
+  from,
 }: {
   occurrence: EventOccurrence;
+  from?: string;
 }) {
   const router = useRouter();
   const { t: translate } = useTranslation();
+  const handleBack = useEventBackNavigation({ from });
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/event/[id].tsx]: Why are we doing this?

File: frontend/apps/mobile/components/home/CategoryCard.tsx [hunk 11]
```diff
@@ -1,26 +1,75 @@
...
 
-const CATEGORY_COLORS: Record<string, string> = {
-  Sport: AppColors.blue,
-  Arts: AppColors.violet,
-  Music: AppColors.pink,
-  Tech: AppColors.emerald,
-  Activity: AppColors.amber,
-  Tutoring: AppColors.danger,
-};
+const CATEGORY_URIS: Record<string, string> = Object.fromEntries(
```
Comment (josh-torre) [inline frontend/apps/mobile/components/home/CategoryCard.tsx]: We should memoize this

File: frontend/apps/mobile/components/home/CategoryCard.tsx [hunk 12]
```diff
@@ -1,26 +1,75 @@
...
+    RNImage.resolveAssetSource(moduleId as number).uri,
+  ]),
+);
 
-export function CategoryCard({
+// Module-level XML content cache — survives remounts, persists for the app's lifetime
+const svgXmlCache = new Map<string, string>();
+
+// Eagerly fetch all SVG content when this module is first imported
+Object.values(CATEGORY_URIS).forEach((uri) => {
```
Comment (josh-torre) [inline frontend/apps/mobile/components/home/CategoryCard.tsx]: Hmm this feels like too much

File: frontend/apps/mobile/hooks/use-event-back-navigation.ts [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import { useRouter } from "expo-router";
+
+type BackNavigationParams = {
+  from?: string;
+};
+
+export function useEventBackNavigation({ from }: BackNavigationParams) {
```
Comment (josh-torre) [inline frontend/apps/mobile/hooks/use-event-back-navigation.ts:7]: Again not sure if we need this

File: frontend/apps/mobile/package.json [hunk 14]
```diff
@@ -63,6 +63,7 @@
     "eslint": "^9.25.0",
     "eslint-config-expo": "~10.0.0",
     "prettier-plugin-tailwindcss": "^0.5.11",
+    "react-native-svg-transformer": "^1.5.3",
```
Comment (josh-torre) [inline frontend/apps/mobile/package.json]: Why do we need this?

File: frontend/apps/mobile/hooks/use-event-back-navigation.ts (hunk 13 above)
Comment (AlenGanopolsky) [reply frontend/apps/mobile/hooks/use-event-back-navigation.ts:7]: I implemented this because it seemed like we needed to refactor how routing logic worked in event category pages. When a user exited from an event after clicking on an event via the event category pages, it would route them back to the homepage since that was initially hardcoded.

File: frontend/apps/mobile/package.json (hunk 14 above)
Comment (AlenGanopolsky) [reply frontend/apps/mobile/package.json]: I was having issues with rendering the event category images initially. Adding/using this library fixed the issue

**Review by adescoteaux1 — DISMISSED** [review]

**Comment by adescoteaux1** [comment]
just fix merge conflicts!

**Review by adescoteaux1 — APPROVED** [review]
lgtm
